src/hmem.c

```diff
@@ -184,3 +184,54 @@ ssize_t ofi_copy_to_hmem_iov(const struct iovec *hmem_iov,
 				     hmem_iov_offset, src, size,
 				     OFI_COPY_BUF_TO_IOV);
 }
+
+int ofi_get_hmem_iface(const void *buf, enum fi_hmem_iface *iface)
```

Comment:
I ask this without reading through the corresponding verbs/rxm PR, but where do you expect to use this? If an application is already giving this information about a buffer, should that not be all a provider needs? If an application is passing in a non-system buffer without setting the right attribute, that will be an erroneous use of the API.

Reply:
I have the requirement of supporting rxm;verbs where the user does not register device memory for local operations. Since an MR is no longer being provided for operations, ofi_get_hmem_iface() is used to rediscover the buffer type.

Note that this capability is being advertised to users by having the FI_HMEM capability set and the FI_MR_HMEM MR mode bit unset.